src/app/shared/contact/contact.component.ts

```diff
@@ -61,8 +65,18 @@ export class ContactComponent implements OnInit, OnDestroy {
       });
   }
 
-  resetForm(): void {
-    this.contactForm.reset();
+  onSubmit(): void {
+    const resultArray: { title: string, value: any }[] = [
+      {title: 'Name', value: this.contactForm.get('name').value},
+      {title: 'Email', value: this.contactForm.get('email').value},
+      {title: 'Topic', value: this.contactForm.get('topic').value},
+      {title: 'Message', value: this.contactForm.get('message').value}
+    ];
+    if (this.contactForm.get('age')) {
+      resultArray.push({title: 'Age', value: this.contactForm.get('age').value});
```

Comment:
this code has bad readability.
My suggestion:
resultArray.push({
    title: 'Age',
    value: this.contactForm.get('age').value
});